client/utils/handleSingleAndDoubleClick.js

```diff
@@ -0,0 +1,30 @@
+import { debounce } from 'lodash'
+
+// Due to limitations of ES6 default strict mode on all files, we cannot bind the
+// 'this' object from calling code. Hence, passing this directly as 'self' (ugly).
```

Comment:
have you tried something like that ?
```js
onClick={() => handleSingleAndDoubleClick(arg)}
```
I find this function a bit confusing. Why does it call methods from self ? callback parameters wouldn't be more clear ?